src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/OptimizedCompletionList.cs

```diff
@@ -174,6 +174,12 @@ private void WriteCompletionItem(JsonWriter writer, CompletionItem completionIte
                     serializer.Serialize(writer, completionItem.Data);
                 }
 
+                if (completionItem is VSCompletionItem vSCompletionItem && vSCompletionItem.VsCommitCharacters is not null)
```

Comment:
Would we be able to lift the completion item commit characters onto the [completion list](https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=/src/product/Protocol/LanguageServer.Protocol.Internal/VSInternalCompletionList.cs&version=GBdevelop&line=62&lineEnd=62&lineStartColumn=9&lineEndColumn=94&lineStyle=plain&_a=contents)? It'd save us from every completion item having its own commit character collection if they're all identical.

Reply:
Maybe I'm missing some context here but that seems dangerous. What if someone has a completion for `somePerviouslyDefiniedVariable` and they complete it with `=` (which is legal and works)? The variable completion here isn't a snippet, so if you complete it with `=` but an attribute existed in that list then you'd get `somePerviouslyDefiniedVariable` instead `somePerviouslyDefiniedVariable=` even though you typed `=`. That's obviously a bit of a contrived example since it assumes an HTML element attribute and C# variable assignment could both be valid responses in the same spot, but that's the kind of scenario I'm worried about.

Reply:
Talked offline and Taylor is correct.